pkg/provisioner/fixture/fixture.go

```diff
@@ -0,0 +1,181 @@
+func (p *fixtureProvisioner) InspectHardware() (dirty bool, err error) {
+	p.log.Info("inspecting hardware", "status", p.host.OperationalStatus())
+
+	if dirty, err = p.ensureExists(); err != nil {
+		return dirty, errors.Wrap(err, "could not inspect hardware")
+	}
+
+	if p.host.OperationalStatus() != metalkubev1alpha1.OperationalStatusInspecting {
+		// The inspection just started.
+		p.log.Info("starting inspection by setting state")
+		p.host.SetOperationalStatus(metalkubev1alpha1.OperationalStatusInspecting)
+		return true, nil
+	}
+
+	// The inspection is ongoing. We'll need to check the fixture
+	// status for the server here until it is ready for us to get the
+	// inspection details. Simulate that for now by creating the
+	// hardware details struct as part of a second pass.
+	if p.host.Status.HardwareDetails == nil {
+		p.log.Info("continuing inspection by setting details")
+		p.host.Status.HardwareDetails =
+			&metalkubev1alpha1.HardwareDetails{
+				RAMGiB: 128,
+				NIC: []metalkubev1alpha1.NIC{
+					metalkubev1alpha1.NIC{
+						MAC:       "some:mac:address",
+						IP:        "192.168.100.1",
+						SpeedGbps: 1,
+					},
+					metalkubev1alpha1.NIC{
+						MAC:       "some:other:mac:address",
+						IP:        "192.168.100.2",
+						SpeedGbps: 1,
+					},
+				},
+				Storage: []metalkubev1alpha1.Storage{
+					metalkubev1alpha1.Storage{
+						SizeGiB: 1024,
+						Info:    "Some information about this disk.",
+					},
+				},
+				CPUs: []metalkubev1alpha1.CPU{
+					metalkubev1alpha1.CPU{
+						Type:     "x86",
+						SpeedGHz: 3,
+					},
+				},
+			}
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// Deprovision prepares the host to be removed from the cluster. It
+// may be called multiple times, and should return true for its dirty
+// flag until the deprovisioning operation is completed.
+func (p *fixtureProvisioner) Deprovision() (dirty bool, retryDelay time.Duration, err error) {
+	p.log.Info("ensuring host is removed")
+
+	// NOTE(dhellmann): In order to simulate a multi-step process,
+	// modify some of the status data structures. This is likely not
+	// necessary once we really have Fixture doing the deprovisioning
+	// and we can monitor it's status.
+
+	if p.host.Status.HardwareDetails != nil {
+		p.log.Info("clearing hardware details")
+		p.host.Status.HardwareDetails = nil
+		return true, deprovisionRequeueDelay, nil
+	}
+
+	if p.host.Status.Provisioning.ID != "" {
+		p.log.Info("clearing provisioning id")
+		p.host.Status.Provisioning.ID = ""
+		return true, deprovisionRequeueDelay, nil
+	}
+
+	return false, 0, nil
+}
+
+// PowerOn ensures the server is powered on independently of any image
+// provisioning operation.
+func (p *fixtureProvisioner) PowerOn() (dirty bool, err error) {
+	p.log.Info("ensuring host is powered on")
+
+	if dirty, err = p.ensureExists(); err != nil {
+		return dirty, errors.Wrap(err, "could not power on host")
+	}
+
+	if p.host.Status.Provisioning.State != "powered on" {
```

Comment:
"powered on" and "powered off" are good candidates for a constant, presumably defined where the `State` struct is defined.

Reply:
Yeah, that'll happen when those methods are doing something useful.